subsys/net/lib/sockets/sockets.c

```diff
@@ -980,8 +980,7 @@ static int zsock_poll_prepare_ctx(struct net_context *ctx,
 {
 	if (pfd->events & ZSOCK_POLLIN) {
 		if (*pev == pev_end) {
-			errno = ENOMEM;
-			return -1;
+			return ENOMEM;
```

Comment:
> return ENOMEM;

Ok, but we should still follow the convention and use negative values for errors (errno's are otherwise positive), so this should be -ENOMEM, etc. elsewhere.

Reply:
I'm aware of the convention, though I got confused by `ioctl` man pages description, namely Linux manual (`man 2 ioctl``), which states:
```
Usually, on success zero is returned.  A few ioctl() requests use the return value as an
output parameter and return a nonnegative value on success.  On error, -1 is returned,
and errno is set appropriately.
```

So according to this description, it should be either `-1` and `errno` or a non-negative value. But I can make it a negative value to make it more Zephyr compliant, no issues with that. And I agree, this could be posted as a separate PR, I'll do that.